packages/opentelemetry-propagator-jaeger/src/JaegerPropagator.ts

```diff
@@ -47,12 +46,15 @@ const UBER_BAGGAGE_HEADER_REGEX = /^uberctx-(.+)/i;
  */
 export class JaegerPropagator implements TextMapPropagator {
   private readonly _jaegerTraceHeader: string;
+  private readonly _jaegerBaggageHeaderPrefix: string;
 
   /**
    * @param {string} [customTraceHeader="uber-trace-id"] - HTTP header to inject\extract trace from.
+   * @param {string} [customBaggageHeaderPrefix="uberctx"] - Baggage header prefix.
    **/
-  constructor(customTraceHeader?: string) {
+  constructor(customTraceHeader?: string, customBaggageHeaderPrefix?: string) {
```

Comment:
Sure do wish we had gone with `constructor(options)` pattern instead of additional fields now :/

Can we maybe use a constructor override to add an options object instead of additional parameters?

Reply:
Hi, I've thought about some kind of `options` param for constructor, but decided to use additional field as using `options` pattern leads to breaking changes.
Ok, will replace 2 fields with object.

Reply:
You can do a type check for backward compatibility. if first argument is of type `string` it's a `customTraceHeader` else interpret it as optional object with optional fields `customTraceHeader: string` and `customBaggageHeaderPrefix: string`

Reply:
Yes, thank you. I can, but I think a code for backward compatibility should be removed in some time and I'm not sure I'll be able to do that in the future.

Reply:
The jaeger propagator is 1.0 so the backwards compatibility needs to be maintained _at least_ until 2.0 which will be quite a long time. I think actually we should just do what you have here and we can consider changing it when we get to 2.0

I was more lamenting the decision we made than anything. i think it's too late to change now.

Reply:
Hi @dyladan. I've found `fields` method inside `JaegerPropagator` and I'm not sure whether I should add  `_jaegerBaggageHeaderPrefix` to returned array or not. Could you, pls, clarify this moment?

Reply:
The fields method returns any headers which may be affected by the propagator. Its main purpose is so the developer can know which fields may need to be cleared if an object is reused multiple times in order to prevent leaking between uses.

Reply:
Got it, thank you. Just pushed new version of code